src/System.Management.Automation/engine/parser/SemanticChecks.cs

```diff
@@ -906,9 +906,12 @@ public override AstVisitAction VisitUnaryExpression(UnaryExpressionAst unaryExpr
             return AstVisitAction.Continue;
         }
 
+        private static bool IsOrderedAttribute(TypeConstraintAst typeConstraint)
+            => typeConstraint.TypeName.FullName.Equals(LanguagePrimitives.OrderedAttribute, StringComparison.OrdinalIgnoreCase);
+
         public override AstVisitAction VisitConvertExpression(ConvertExpressionAst convertExpressionAst)
         {
-            if (convertExpressionAst.Type.TypeName.FullName.Equals(LanguagePrimitives.OrderedAttribute, StringComparison.OrdinalIgnoreCase))
+            if (IsOrderedAttribute(convertExpressionAst.Type))
```

Comment:
Once the other use of `IsOrderedAttribute` is reverted below, this can be reverted too.